controller/gui/client.js

```diff
@@ -65,3 +61,57 @@ customElements.define(
   },
   { extends: 'input' }
 )
+
+/* Form web component preventing more than one submission.
+ *
+ * Disable submit input inside the form after the first submission.
+ */
+customElements.define(
+  'disable-after-submit',
+  class extends HTMLFormElement {
+    constructor() {
+      super()
+
+      const form = this
+      const button = form.querySelector('input[type=submit]')
+
+      const buttonParent = installParent(button, document.createElement('span'))
+      buttonParent.style = `
+        position: relative;
+        height: fit-content;
+      `
+
+      const spinnerParent = document.createElement('div')
+      spinnerParent.style = `
+        display: flex;
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+      `
+
+      const spinner = document.createElement('span')
+      spinner.className = 'd-Spinner'
+      spinnerParent.appendChild(spinner)
+
+      form.addEventListener('submit', function() {
+        button.disabled = true
+        button.style.color = 'transparent'
```

Comment:
This style could be moved to the css file, since we're already styling disabled buttons there.

Reply:
That’s right, it works with this usage. Now, if we think about another usage: disabling a button without showing a loader, we still would like to show the button content.

So, either we keep this manual style, or we introduce another style. I suggest to stick to update the style here, since it’s the only place, and we can think about using a style if this is used in another scenario, or for more complex rules in the future.

Reply:
:+1: